hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -80,8 +83,9 @@ public void sendMessage(final Message message) {
         }
     }
 
-    public void updateSubscription(Subscription newSubscription) {
+    public synchronized void updateSubscription(Subscription newSubscription) {
         this.subscription = newSubscription;
+        this.messageSender = messageSenderFactory.create(newSubscription);
```

Comment:
I think my comments made no sense because component that you replaced are responsible for single message delivery